spring-cloud-gcp-pubsub/src/main/java/com/google/cloud/spring/pubsub/support/DefaultSubscriberFactory.java

```diff
@@ -183,13 +215,15 @@ public Subscriber createSubscriber(String subscriptionName, MessageReceiver rece
 		Subscriber.Builder subscriberBuilder = Subscriber.newBuilder(
 				PubSubSubscriptionUtils.toProjectSubscriptionName(subscriptionName, this.projectId), receiver);
 
+		PubSubConfiguration.Subscriber subscriberProperties = this.pubSubConfiguration
```

Comment:
What happens if this is null? (it will be if the deprecated `DefaultSubscriberFactory` constructor is called). `Subscriber.Builder` has its own default, which it's probably fine to fall back on. But you definitely want to gate `subscriberBuilder.setExecutorProvider(getExecutorProvider(subscriberProperties, subscriptionName));` on the presence of at least `subscriberProperties`.

Reply:
Done. Added a `null` check for `this.pubSubConfiguration` when we call `subscriberBuilder.setExecutorProvider(getExecutorProvider(subscriberProperties, subscriptionName))` to prevent a `NullPointerException` from occurring.